safe-global/safe-wallet-web
# Feat: dynamically block sactioned addresses [SW-11]

## What it solves

Resolves: https://www.notion.so/safe-global/Dynamic-OFAC-addresses-blocking-b4f02f154d8140e6aeeaea9947c69090

## How this PR fixes it
- Uses a chainalysis contract to check if an address is blocked: https://go.chainalysis.com/chainalysis-oracle-docs.html. 
- The RPC request is done with RTK query, which stores the result for each address in state.
- The blocking is for the swaps feature. It will check based on the wallet address, safe address, and recipient address (if given).

## How to test it

Blocked wallet address:
- Impersonate an address on the OFAC block list.
- Try to use the swap feature.
- The blocked message should show.

Blocked recipient address:
- Using a non blocked wallet address, go to the swaps feature.
- Enable custom recipients and provide a blocked address.
- The blocked message should show. 

Blocked safe address:
- Not sure how we can test this one, I don't think are aware of any safe addresses on the block list.

## Checklist
